build fails if `git` is configured to show gpg signatures

### Bug Description

While building from HEAD (3aa73fd1cb0f39c3993bff2913f72c7d03c58c13), it failed for me with:

```shell
[ 12%] Built target mixxx-qml-mixxxcontrols_autogen                                                                   
In file included from /home/vbatts/src/mixxx/src/util/gitinfostore.cpp:4:                                             
/home/vbatts/src/mixxx/build/src/gitinfo.h:13:25: warning: missing terminating " character                                                                                                                                                   
   13 | #define GIT_COMMIT_DATE "gpg: Signature made Thu 21 Mar 2024 01:37:29 PM EDT                                  
      |                         ^                                                                                     
/home/vbatts/src/mixxx/build/src/gitinfo.h:15:9: warning: missing terminating ' character                                                                                                                                                    
   15 | gpg: Can't check signature: No public key                                                                                                                                                                                            
      |         ^                                                                                                                                                                                                                            
/home/vbatts/src/mixxx/build/src/gitinfo.h:15:9: error: missing terminating ' character                               
   15 | gpg: Can't check signature: No public key                                                                     
      |         ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~                                                                     
/home/vbatts/src/mixxx/build/src/gitinfo.h:16:26: warning: missing terminating " character                            
   16 | 2024-03-21T18:37:29+01:00"                                                                                                                                                                                                           
      |                          ^                                                                                    
/home/vbatts/src/mixxx/build/src/gitinfo.h:16:26: error: missing terminating " character                                                                                                                                                     
/home/vbatts/src/mixxx/build/src/gitinfo.h:13:25: error: missing terminating " character                              
   13 | #define GIT_COMMIT_DATE "gpg: Signature made Thu 21 Mar 2024 01:37:29 PM EDT                                  
      |                         ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~                                                                                                                                                         
/home/vbatts/src/mixxx/build/src/gitinfo.h:13:25: note: in definition of macro ‘GIT_COMMIT_DATE’                                                                                                                                             
   13 | #define GIT_COMMIT_DATE "gpg: Signature made Thu 21 Mar 2024 01:37:29 PM EDT                                  
      |                         ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~                                                                                                                                                         
/home/vbatts/src/mixxx/build/src/gitinfo.h:14:1: error: ‘gpg’ does not name a type                                    
   14 | gpg:                using RSA key B5690EEEBB952194                                                            
      | ^~~                                                                                                           
/home/vbatts/src/mixxx/src/util/gitinfostore.cpp: In static member function ‘static const char* GitInfoStore::date()’:
/home/vbatts/src/mixxx/src/util/gitinfostore.cpp:18:5: error: return-statement with no value, in function returning ‘const char*’ [-fpermissive]
   18 |     return GIT_COMMIT_DATE;                                                                                                                                                                                                          
      |     ^~~~~~                                                                                                    
make[2]: *** [CMakeFiles/mixxx-gitinfostore.dir/build.make:76: CMakeFiles/mixxx-gitinfostore.dir/src/util/gitinfostore.cpp.o] Error 1
```

Looking at `build/src/gitinfo.h`:
```c
vbatts@jungle:~/src/mixxx/build$ cat src/gitinfo.h 
#pragma once
// This file contains the exact Mixxx git version and commit date. Since this
// changes on every single commit (or even if just the dirty flag changes), all
// files that include this will be recompile very often. Therefore, this should
// only be included from versionstore.cpp.

#ifndef GIT_INFO
    #error "Include only from gitinfostore.cpp and mixxx.rc.include to avoid unnecessary recompile on every commit."
#endif

#define GIT_BRANCH ""
#define GIT_DESCRIBE ""
#define GIT_COMMIT_DATE "gpg: Signature made Thu 21 Mar 2024 01:37:29 PM EDT
gpg:                using RSA key B5690EEEBB952194
gpg: Can't check signature: No public key
2024-03-21T18:37:29+01:00"
/* #undef GIT_COMMIT_YEAR */
/* #undef GIT_COMMIT_COUNT */
/* #undef GIT_DIRTY */
```

it surely is malformed.


## investigation

On my machine I have `~/.gitconfig`  configured with:
```ini
[log]
  showSignature = true
```

Which is causing this additional information to be included in this header, when the [`./cmake/modules/GitInfo.cmake` runs  `git describe`](https://github.com/mixxxdj/mixxx/blob/3aa73fd1cb0f39c3993bff2913f72c7d03c58c13/cmake/modules/GitInfo.cmake#L4).
I looked whether there was a flag to be used like `--no-show-signature` or `--show-signature=false`, but there is not.


## workaround

In this source clone for mixxx, add to `./.git/config`:
```ini
[log]
  showSignature = false
```

Perhaps the setup scripts could just add this.

### Version

_No response_

### OS

_No response_

--- a/cmake/modules/GitInfo.cmake
+++ b/cmake/modules/GitInfo.cmake
@@ -54,7 +54,7 @@ endif()
 # Get the current commit date
 if(NOT GIT_COMMIT_DATE)
   execute_process(
-    COMMAND git show --quiet --format=%cI --date=short
+    COMMAND git show --quiet --format=%cI --date=short --no-show-signature
     WORKING_DIRECTORY "${CMAKE_CURRENT_SOURCE_DIR}"
     OUTPUT_VARIABLE GIT_COMMIT_DATE
     OUTPUT_STRIP_TRAILING_WHITESPACE

